File: frontend/School_Mangement/client/types/admin/teacher.ts
export type Teacher = {
    id: number;

    employeeId: string;

    firstName: string;

    lastName: string;

    name: string;

    email: string;

    phone: string;

    qualification: string;

    specialization: string;

    joiningDate: string;

    dateOfBirth: string;

    status: "ACTIVE" | "INACTIVE";

    role: string;
};
export interface TeacherFormData {
    firstName: string;
    lastName: string;
    email: string;
    phone: string;
    qualification: string;
    specialization: string;
    dateOfBirth: string;
    status: string;
    role: string;
}

export interface TeacherStats {
    totalTeacher: number;
    male: number;
    female: number;
    toadayPresent: number;
}

File: backend/Smart_Erp/src/main/java/com/Smart/Erp/Entity/Teacher.java
package com.Smart.Erp.Entity;

import com.Smart.Erp.Enum.Status;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "teachers")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Teacher {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String employeeId;

    private String firstName;
    private String lastName;

    private String email;
    private String phone;

    private String qualification;
    private String specialization;

    private LocalDate joiningDate;

    private LocalDate dateOfBirth;

    private String gender;

    @Enumerated(EnumType.STRING)
    private Status status;

    @OneToOne
    @JoinColumn(name = "user_id")
    private User user;

    @ManyToMany(fetch = FetchType.LAZY)
    @JoinTable(
            name = "teacher_subject",
            joinColumns = @JoinColumn(name = "teacher_id"),
            inverseJoinColumns = @JoinColumn(name = "subject_id")
    )
    private List<Subject> subjects = new ArrayList<>();

    @OneToMany(mappedBy = "classTeacher")
    private List<Section> classTeacherSections = new ArrayList<>();
}
